GCC builds should always build aws sdk in release mode

GCC builds should always build aws sdk in release mode. Debug builds cause asm error in compilation:

```
/home/seth/git/TileDB/build/externals/src/ep_awssdk-build/.deps/build/src/AwsChecksums/source/intel/crc32c_sse42_asm.c: In function 's_crc32c_sse42_clmul_256':
/home/seth/git/TileDB/build/externals/src/ep_awssdk-build/.deps/build/src/AwsChecksums/source/intel/crc32c_sse42_asm.c:61:5: error: 'asm' operand has impossible constraints
   61 |     asm volatile("enter_256_%=:"
      |     ^~~
gmake[8]: *** [CMakeFiles/aws-checksums.dir/build.make:141: CMakeFiles/aws-checksums.dir/source/intel/crc32c_sse42_asm.c.o] Error 1
gmake[7]: *** [CMakeFiles/Makefile2:398: CMakeFiles/aws-checksums.dir/all] Error 2
gmake[6]: *** [Makefile:141: all] Error 2
gmake[5]: *** [CMakeFiles/AwsChecksums.dir/build.make:113: build/src/AwsChecksums-stamp/AwsChecksums-build] Error 2
gmake[4]: *** [CMakeFiles/Makefile2:148: CMakeFiles/AwsChecksums.dir/all] Error 2
gmake[3]: *** [Makefile:84: all] Error 2
CMake Error at CMakeLists.txt:193 (message):
  Failed to build third-party libraries
```

Upstream says to use release builds with gcc:
https://github.com/awslabs/aws-checksums/issues/8
https://github.com/awslabs/aws-checksums/commit/ebcae257a416de8ce0f701b49861d0406a592fe6


Credit to @ihnorton for debugging the issue.


Previously aws used release builds always but windows doesn't like mixed builds (release aws, debug tiledb). Thus we need to only force release builds for gcc.

--- a/cmake/Modules/FindAWSSDK_EP.cmake
+++ b/cmake/Modules/FindAWSSDK_EP.cmake
@@ -69,6 +69,15 @@ if (NOT AWSSDK_FOUND)
       list(APPEND DEPENDS ep_zlib)
     endif()
 
+    # Set AWS cmake build to use specified build type except for gcc
+    # For aws sdk and gcc we must always build in release mode
+    # See https://github.com/TileDB-Inc/TileDB/issues/1351 and
+    # https://github.com/awslabs/aws-checksums/issues/8
+    set(AWS_CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE})
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "GNU")
+        set(AWS_CMAKE_BUILD_TYPE "Release")
+    endif()
+
     # Work around for: https://github.com/aws/aws-sdk-cpp/pull/1187
     # AWS SDK builds it's "aws-common" dependencies using `execute_process` to run cmake directly,
     # and does not pass the CMAKE_GENERATOR PLATFORM, so those projects default to Win32 builds,
@@ -87,7 +96,7 @@ if (NOT AWSSDK_FOUND)
       URL "https://github.com/aws/aws-sdk-cpp/archive/1.7.108.zip"
       URL_HASH SHA1=d07bae3fe924008b3117936963456dd314787abf
       CMAKE_ARGS
-        -DCMAKE_BUILD_TYPE=${CMAKE_BUILD_TYPE}
+        -DCMAKE_BUILD_TYPE=${AWS_CMAKE_BUILD_TYPE}
         -DENABLE_TESTING=OFF
         -DBUILD_ONLY=s3\\$<SEMICOLON>core
         -DBUILD_SHARED_LIBS=OFF

